src/filters/medianFilter.ts

```diff
@@ -0,0 +1,74 @@
+import { xMedian } from 'ml-spectra-processing';
+
+import { Image } from '../Image';
+import checkProcessable from '../utils/checkProcessable';
+import { getBorderInterpolation, BorderType } from '../utils/interpolateBorder';
+
+export interface MedianFilterOptions {
+  /**
+   * Type of border algorithm to interpolate from.
+   *
+   * @default 'reflect101'
+   */
+  borderType: BorderType;
+  /**
+   * Value of border.
+   */
+  borderValue: number;
+  /**
+   * The radius of the cell to extract xMedian from. Must be odd.
+   *
+   * @default 1
+   */
+  cellSize: number;
+}
+/**
+ * Apply a median filter to the image.
```

Comment:
Please don't paraphrase the function's name. It should describe what the function does in 2-3 sentences, not state what the function is.